lib/content-services/src/lib/tag/tags-creator/tags-creator.component.ts

```diff
@@ -0,0 +1,379 @@
+/*!
+ * @license
+ * Copyright 2019 Alfresco Software, Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { TagEntry, TagPaging } from '@alfresco/js-api';
+import {
+    Component,
+    ElementRef,
+    EventEmitter,
+    HostBinding,
+    Input,
+    OnDestroy,
+    OnInit,
+    Output,
+    ViewChild,
+    ViewEncapsulation
+} from '@angular/core';
+import { FormControl, Validators } from '@angular/forms';
+import { debounce, distinctUntilChanged, finalize, first, map, takeUntil, tap } from 'rxjs/operators';
+import { EMPTY, forkJoin, Observable, Subject, timer } from 'rxjs';
+import { NotificationService } from '@alfresco/adf-core';
+import { TagsCreatorMode } from './tags-creator-mode';
+import { MatSelectionListChange } from '@angular/material/list';
+import { TagService } from '../services/tag.service';
+
+interface TagNameControlErrors {
+    duplicatedExistingTag?: boolean;
+    duplicatedAddedTag?: boolean;
+    emptyTag?: boolean;
+    required?: boolean;
+}
+
+const DEFAULT_TAGS_SORTING = {
+    orderBy: 'tag',
+    direction: 'asc'
+};
+
+@Component({
+    selector: 'adf-tags-creator',
+    templateUrl: './tags-creator.component.html',
+    styleUrls: ['./tags-creator.component.scss'],
+    encapsulation: ViewEncapsulation.None
+})
+export class TagsCreatorComponent implements OnInit, OnDestroy {
+    @Input()
+    mode: TagsCreatorMode;
+
+    @Input()
+    disabledTagsRemoving = false;
+
+    @Input()
+    set tags(tags: string[]) {
+        this._tags = [...tags];
+        this._existingTagsLoading = true;
```

Comment:
maybe this one should be set inside the "loadTags" method? to avoid confusions

Reply:
I addressed that comment partially. 
It must be where it is because please notice that it is used in couple places and in one of them we have timer 300 for debouncing input purpose and we would like to display spinner before timing ends but loading is happening after timing. 

I agree that "existingTagsLoading" sounds like something what should be inside loadTags so to resolve that problem I renamed that boolean to something what should be fine for places where it is set.